typing/typedecl.ml

```diff
@@ -55,11 +55,23 @@ type error =
   | Cannot_unbox_or_untag_type of native_repr_kind
   | Deep_unbox_or_untag_attribute of native_repr_kind
   | Bad_immediate_attribute
+  | Bad_unboxed_attribute of string
+  | Wrong_unboxed_type_float
+  | Boxed_and_unboxed
 
 open Typedtree
 
 exception Error of Location.t * error
 
+let get_unboxed_from_attributes sdecl =
+  let unboxed = Builtin_attributes.has_unboxed sdecl.ptype_attributes in
+  let boxed = Builtin_attributes.has_boxed sdecl.ptype_attributes in
+  if boxed && unboxed then raise (Error(sdecl.ptype_loc, Boxed_and_unboxed))
+  else if boxed then { unboxed = false; default = false }
+  else if unboxed then { unboxed = true; default = false }
+  else if !Clflags.unboxed_types then { unboxed = true; default = true }
+  else { unboxed = false; default = true }
+
```

Comment:
Perhaps write this with a pattern matching on the three relevant Booleans (with a comment to prevent clever developers from writing `{unboxed=!Clflags.unboxed_types; default=true}`, which would loose the nice sharing property).